[Release-1.29] Release Notes (#17119)
istio/istio.io: content/en/news/releases/1.29.x/announcing-1.29/change-notes/index.md

---
title: Istio 1.29.0 Change Notes
linktitle: 1.29.0
subtitle: Minor Release
description: Istio 1.29.0 release notes.
publishdate: 2026-02-12
release: 1.29.0
weight: 10
aliases:
    - /news/announcing-1.29.0
---

## Traffic Management

- **Promoted** the `cni.ambient.dnsCapture` value to default to `true`.
This enables DNS proxying for workloads in ambient mesh by default, improving security and performance while enabling a number of features. This can be disabled explicitly or with `compatibilityVersion=1.24`.
Note: only new pods will have DNS enabled. To enable DNS for existing pods, pods must be manually restarted, or the iptables reconciliation feature must be enabled with `--set cni.ambient.reconcileIptablesOnStartup=true`.

- **Promoted** `cni.ambient.reconcileIptablesOnStartup` to default to `true`.
This enables automatic reconciliation of iptables/nftables rules for existing ambient pods when the `istio-cni` DaemonSet is upgraded,
eliminating the need to manually restart pods to receive updated networking configuration.

> This can be disabled explicitly or by using `compatibilityVersion=1.28`.

Reviewer: ```suggestion
- **Promoted** `cni.ambient.reconcileIptablesOnStartup` to default to `true`.
  This enables automatic reconciliation of iptables/nftables rules for existing ambient pods when the `istio-cni` DaemonSet is upgraded,
  eliminating the need to manually restart pods to receive updated networking configuration.
  This can be disabled explicitly or by using `compatibilityVersion=1.28`.
```

This feature currently remains off by default and can be turned on with the `ENABLE_GATEWAY_API_INFERENCE_EXTENSION` environment variable.
([usage]( https://istio.io/latest/docs/tasks/traffic-management/ingress/gateway-api-inference-extension/)) ([Issue #58533](https://github.com/istio/istio/issues/58533))

- **Added** support for Istio locality label `topology.istio.io/locality`, which takes precedence over `istio-locality`.

- **Added** an option, `gateway.istio.io/tls-cipher-suites`, to specify the custom cipher suites on a Gateway. The value is a comma separated list of cipher suites.
  ([Issue #58366](https://github.com/istio/istio/issues/58366))

- **Added** alpha support for a baggage-based telemetry system for ambient mesh. Users of multinetwork
ambient will want to enable this feature via the `AMBIENT_ENABLE_BAGGAGE` pilot environment variable so that
metrics for cross-network traffic are properly attributed with source and destination labels. Note that
ztunnel already sends baggage in requests; this feature augments that functionality with waypoint-generated
baggage as well. As such, this feature is off by default for waypoints and on by default

> in ztunnels (configurable via the `ENABLE_RESPONSE_BAGGAGE` environment variable in ztunnel).

Reviewer: ```suggestion
- **Added** alpha support for a baggage-based telemetry system for ambient mesh. Users of multinetwork
  ambient will want to enable this feature via the `AMBIENT_ENABLE_BAGGAGE` pilot environment variable so that
  metrics for cross-network traffic are properly attributed with source and destination labels. Note that
  ztunnel already sends baggage in requests; this feature augments that functionality with waypoint-generated
  baggage as well. As such, this feature is off by default for waypoints and on by default
  in ztunnels (configurable via the `ENABLE_RESPONSE_BAGGAGE` environment variable in ztunnel).
```

> - **Added** logic to designate a Workload Discovery (WDS) Service as canonical. A canonical WDS Service is used by ztunnel during name resolution unless another WDS Service in the same namespace as the client exists to override it. A canonical service will be configured from either (1) a Kubernetes Service resource or (2) the oldest Istio ServiceEntry resource that specifies that hostname.

Reviewer: ```suggestion
- **Added** logic to designate a Workload Discovery (WDS) Service as canonical.
  A canonical WDS Service is used by ztunnel during name resolution unless another WDS Service
  in the same namespace as the client exists to override it. A canonical service will be configured
  from either (1) a Kubernetes `Service` resource or (2) the oldest Istio `ServiceEntry` resource that
  specifies that hostname.
```

  ([Issue #58576](https://github.com/istio/istio/pull/58576))

- **Added** a new feature flag `DISABLE_TRACK_REMAINING_CB_METRICS` to control circuit breaker remaining metrics tracking.
When set to `false` (default), circuit breaker remaining metrics will not be tracked, improving performance.
When set to `true`, circuit breaker remaining metrics will be tracked (legacy behavior).
This feature flag will be removed in a future release.

- **Added** support for `LEAST_REQUEST` load balancing policy in gRPC proxyless clients.

- **Added** support for circuit breaking (`http2MaxRequests`) in gRPC proxyless clients.

- **Added** support for wildcard hosts in ServiceEntries with `DYNAMIC_DNS` resolution
for TLS hosts. The TLS protocol implies that connections will be routed based on the
request's SNI (from the TLS handshake) without terminating the TLS connection to
inspect the Host header for routing. The implementation relies on an alpha API
and has significant security implications (i.e., SNI spoofing). Therefore, this
feature is disabled by default and can be enabled by setting the feature flag
`ENABLE_WILDCARD_HOST_SERVICE_ENTRIES_FOR_TLS` to `true`. Please consider using

> this feature carefully and only with trusted clients.  ([Issue #54540](https://github.com/istio/istio/issues/54540))

Reviewer: ```suggestion
- **Added** support for wildcard hosts in `ServiceEntry` resources with `DYNAMIC_DNS` resolution
  for TLS hosts. The TLS protocol implies that connections will be routed based on the
  request's SNI (from the TLS handshake) without terminating the TLS connection to
  inspect the Host header for routing. The implementation relies on an alpha API
  and has significant security implications (i.e., SNI spoofing). Therefore, this
  feature is disabled by default and can be enabled by setting the feature flag
  `ENABLE_WILDCARD_HOST_SERVICE_ENTRIES_FOR_TLS` to `true`. Please consider using
  this feature carefully and only with trusted clients.
  ([Issue #54540](https://github.com/istio/istio/issues/54540))
```

> - **Fixed** an issue where sidecars tried to route requests to ambient E/W gateways incorrectly.

Reviewer: ```suggestion
- **Fixed** an issue where sidecars tried to route requests to ambient east/west gateways incorrectly.
```

  ([Issue #57878](https://github.com/istio/istio/issues/57878))

- **Fixed** Istio CNI node agent startup failure in MicroK8s environments when using ambient mode with nftables backend.
  ([Issue #58185](https://github.com/istio/istio/issues/58185))

> - **Fixed** an issue where InferencePool configurations were lost during VirtualService merging when multiple HTTPRoute referencing different InferencePools were attached to the same Gateway.  ([Issue #58392](https://github.com/istio/istio/issues/58392))

Reviewer: ```suggestion
- **Fixed** an issue where `InferencePool` configurations were lost during `VirtualService` merging when multiple `HTTPRoute` referencing different `InferencePool`s were attached to the same Gateway.
  ([Issue #58392](https://github.com/istio/istio/issues/58392))
```

  ([Issue #58731](https://github.com/istio/istio/pull/58731))

- **Fixed** an issue where warmup aggression is not aligned with envoy configuration.
  ([Issue #3395](https://github.com/istio/api/issues/3395))

- **Fixed** an issue where ingress gateways in ambient multi-cluster did not route requests to exposed remote backends. Also, a new feature flag `AMBIENT_ENABLE_MULTI_NETWORK_INGRESS` has been added and it's `true` by default. If the user wants to keep the old behavior, it can be set to `false`.

- **Fixed** an issue causing the ambient multicluster cluster registry to become unstable periodically, leading to incorrect configuration being pushed to proxies.

- **Fixed** an issue where the overload manager resource monitor for global downstream max connections
was set to the maximum integer value and could not be configured via Runtime Flags.
Users can now configure the global downstream max connections limit via proxy metadata `ISTIO_META_GLOBAL_DOWNSTREAM_MAX_CONNECTIONS`.
The runtime flag `overload.global_downstream_max_connections` is still honored if specified for backwards compatibility but is deprecated in favor

> of this new approach using proxy metadata.

Reviewer: ```suggestion
- **Fixed** an issue where the overload manager resource monitor for global downstream max connections
  was set to the maximum integer value and could not be configured via Runtime Flags.
  Users can now configure the global downstream max connections limit via proxy metadata `ISTIO_META_GLOBAL_DOWNSTREAM_MAX_CONNECTIONS`.
  The runtime flag `overload.global_downstream_max_connections` is still honored if specified for backwards compatibility but is deprecated in favor
  of this new approach using proxy metadata.
```

- **Added** an experimental feature to allow dry-run AuthorizationPolicy in ztunnel. This feature will be disabled by default. See the Upgrade Note for details.
 ([usage]( https://istio.io/latest/docs/tasks/security/authorization/authz-dry-run/)) ([Issue #1933](https://github.com/istio/api/pull/1933))

- **Added** support to block CIDRs in JWKS URIs when fetching public keys for JWT validation.
If any resolved IP from a JWKS URI matches a blocked CIDR, Istio will skip fetching the public key
and use a fake JWKS instead to reject requests with JWT tokens.

- **Added** a retry mechanism when checking if a pod is ambient enabled in istio-cni.
This is to address potential transient failures resulting in potential mesh bypassing. This feature
is disabled by default and can be enabled by setting `ambient.enableAmbientDetectionRetry` in the
`istio-cni` chart.

- **Added** namespace-based authorization for debug endpoints on port 15014.
Non-system namespaces restricted to config_dump/ndsz/edsz endpoints and same-namespace proxies only.
Disable with `ENABLE_DEBUG_ENDPOINT_AUTH=false` if needed for compatibility.

- **Added** optional NetworkPolicy deployment for ztunnel.

> You can set `global.networkPolicy.enabled=true` to deploy a default NetworkPolicy for ztunnel.  ([Issue #56877](https://github.com/istio/api/issues/56877))

Reviewer: ```suggestion
- **Added** optional `NetworkPolicy` deployment for ztunnel.

  You can set `global.networkPolicy.enabled=true` to deploy a default `NetworkPolicy` for ztunnel.
  ([Issue #56877](https://github.com/istio/api/issues/56877))
```

- **Fixed** resource annotation validation to reject newlines and control characters that could inject containers into pod specs via template rendering.
  ([Issue #58889](https://github.com/istio/istio/issues/58889))

## Telemetry

- **Added** `statsCompression` option in `proxyConfig` to allow global configuration of HTTP compression for the Envoy stats endpoint exposing its metrics. This is enabled by default, offering `brotli`, `gzip` and `zstd` depending on the `Accept-Header` sent by the client.
**Deprecated** the `sidecar.istio.io/statsCompression` annotation, which is replaced by the `statsCompression` `proxyConfig` option. Per-pod overrides are still possible via `proxy.istio.io/config` annotation.
  ([Issue #48051](https://github.com/istio/istio/issues/48051))

- **Added** source and destination workload identification to waypoint proxy traces.
Waypoint proxies now include `istio.source_workload`, `istio.source_namespace`, `istio.destination_workload`, `istio.destination_namespace` and
other source peer tags in trace spans, matching the observability capabilities of sidecar proxies.
  ([Issue #58348](https://github.com/istio/istio/issues/58348))

> - **Added** support for `Formatter` type custom tag in Telemetry API.

Reviewer: ```suggestion
- **Added** support for `Formatter` type custom tag the in Telemetry API.
```

to true for pilot.
  ([Issue #58794](https://github.com/istio/istio/issues/58794)),([Issue #58476](https://github.com/istio/istio/issues/58476))

## Installation

- **Updated** `istiod` to set `GOMEMLIMIT` to 90% of the memory limit (previously 100%) to reduce the risk of OOM kills.
This is now handled automatically via the `automemlimit` library. Users can override this by setting the `GOMEMLIMIT`
environment variable directly, or adjust the ratio using the `AUTOMEMLIMIT` environment variable (e.g., `AUTOMEMLIMIT=0.85` for 85%).

- **Updated** Kiali addon to version `v2.21.0`.

- **Added** support for filtering resources that Pilot will watch, based on the environment variable `PILOT_IGNORE_RESOURCES`.
This variable is a comma-separated list of resources and prefixes that should be ignored by the Istio CRD Watcher.
If there is a need to explicitly include a resource, even when it is on the ignore list, this can be done
using the variable `PILOT_INCLUDE_RESOURCES`.
This feature enables administrators to deploy Istio as a Gateway API-only controller, ignoring mesh resources,

> or to deploy Istio with support only for Gateway API HTTPRoute (e.g., GAMMA support).

Reviewer: ```suggestion
- **Added** support for filtering resources that Pilot will watch, based on the environment variable `PILOT_IGNORE_RESOURCES`.

  This variable is a comma-separated list of resources and prefixes that should be ignored by the Istio CRD watcher.
  If there is a need to explicitly include a resource, even when it is on the ignore list, this can be done
  using the variable `PILOT_INCLUDE_RESOURCES`.

  This feature enables administrators to deploy Istio as a Gateway API-only controller, ignoring mesh resources,
  or to deploy Istio with support only for Gateway API `HTTPRoute` (e.g., GAMMA support).
```
